src/main/java/org/junit/internal/ComparisonCriteria.java

```diff
@@ -56,27 +68,31 @@ public void arrayEquals(String message, Object expecteds, Object actuals)
                 }
             }
         }
-    }
 
-    private boolean isArray(Object expected) {
-        return expected != null && expected.getClass().isArray();
+        if (actualsLength != expectedsLength) {
+            Object expected = getArrayElementOrSentinel(expecteds, expectedsLength, prefixLength);
+            Object actual = getArrayElementOrSentinel(actuals, actualsLength, prefixLength);
+            try {
+                Assert.assertEquals(expected, actual);
```

Comment:
@kcooney I need to do an assertEquals to get the right failure message. I could use fail, but then i would have to construct the failure message here, and that would duplicate the code in assertEquals.